### ISSUE TITLE
Error: Unknown Scheme - Grammarly Extension Failing to Start in VSCode

### ISSUE THREAD

**Issue by segunak** [znck/grammarly#386]
### Discussed in https://github.com/znck/grammarly/discussions/385

<div type='discussions-op-text'>

<sup>Originally posted by **KedneckInc** September  8, 2023</sup>
Environment:

- Windows: 11.?
- Visual Studio Code: 1.82.0

What has been tried:

- Latest PreRelease,
- Latest Released,
- Released Version 0.20.0.

The crazy thing is it was working just a day or three ago!  I'm afraid I didn't note which version it was running at that time.

The attempt to start the language server gives results like this.  This sample is from 0.20.0:

```
`Ready!
Initialized!
open file:///c%3A/msys64/REDACTED.md
create text checking session for "file:///c%3A/msys64/REDACTED.md" with {
  "documentDialect": "auto-text",
  "documentDomain": "creative",
  "suggestions": {
    "ConjunctionAtStartOfSentence": false,
    "Fluency": true,
    "InformalPronounsAcademic": false,
    "MissingSpaces": true,
    "NounStrings": true,
    "NumbersBeginningSentences": true,
    "NumbersZeroThroughTen": true,
    "OxfordComma": true,
    "PassiveVoice": false,
    "PersonFirstLanguage": false,
    "PossiblyBiasedLanguageAgeRelated": false,
    "PossiblyBiasedLanguageDisabilityRelated": false,
    "PossiblyBiasedLanguageFamilyRelated": false,
    "PossiblyBiasedLanguageGenderRelated": false,
    "PossiblyBiasedLanguageHumanRights": false,
    "PossiblyBiasedLanguageHumanRightsRelated": false,
    "PossiblyBiasedLanguageLgbtqiaRelated": false,
    "PossiblyBiasedLanguageRaceEthnicityRelated": false,
    "PossiblyPoliticallyIncorrectLanguage": false,
    "PrepositionAtTheEndOfSentence": true,
    "PunctuationWithQuotation": true,
    "ReadabilityFillerwords": true,
    "ReadabilityTransforms": true,
    "SentenceVariety": true,
    "SpacesSurroundingSlash": true,
    "SplitInfinitive": true,
    "StylisticFragments": false,
    "UnnecessaryEllipses": false,
    "Variety": true,
    "Vocabulary": true
  }
} 
node:internal/deps/undici/undici:11413
    Error.captureStackTrace(err, this);
          ^

TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11413:11) {
  cause: Error: unknown scheme
      at makeNetworkError (node:internal/deps/undici/undici:6768:35)
      at schemeFetch (node:internal/deps/undici/undici:10875:18)
      at node:internal/deps/undici/undici:10748:26
      at mainFetch (node:internal/deps/undici/undici:10765:11)
      at fetching (node:internal/deps/undici/undici:10722:7)
      at fetch2 (node:internal/deps/undici/undici:10600:20)
      at Object.fetch (node:internal/deps/undici/undici:11411:18)
      at fetch (node:internal/process/pre_execution:248:25)
      at c:\Users\spind\.vscode\extensions\znck.grammarly-0.20.0\dist\server\index.node.js:34:77333
      at c:\Users\spind\.vscode\extensions\znck.grammarly-0.20.0\dist\server\index.node.js:34:77553 {
    [cause]: undefined
  }
}

Node.js v18.15.0
[Error - 12:48:50 PM] Connection to server got closed. Server will not be restarted.
[Error - 12:48:50 PM] Request textDocument/codeAction failed.
Error: Connection got disposed.
	at Object.dispose (c:\Users\spind\.vscode\extensions\znck.grammarly-0.20.0\dist\extension\index.node.js:30:4048)
	at Object.dispose (c:\Users\spind\.vscode\extensions\znck.grammarly-0.20.0\dist\extension\index.node.js:35:9520)
	at os.handleConnectionClosed (c:\Users\spind\.vscode\extensions\znck.grammarly-0.20.0\dist\extension\index.node.js:37:9054)
	at os.handleConnectionClosed (c:\Users\spind\.vscode\extensions\znck.grammarly-0.20.0\dist\extension\index.node.js:40:12095)
	at r (c:\Users\spind\.vscode\extensions\znck.grammarly-0.20.0\dist\extension\index.node.js:37:8736)
	at Zs.invoke (c:\Users\spind\.vscode\extensions\znck.grammarly-0.20.0\dist\extension\index.node.js:3:9583)
	at Kr.fire (c:\Users\spind\.vscode\extensions\znck.grammarly-0.20.0\dist\extension\index.node.js:3:10348)
	at Rt (c:\Users\spind\.vscode\extensions\znck.grammarly-0.20.0\dist\extension\index.node.js:4:10852)
	at Zs.invoke (c:\Users\spind\.vscode\extensions\znck.grammarly-0.20.0\dist\extension\index.node.js:3:9583)
	at Kr.fire (c:\Users\spind\.vscode\extensions\znck.grammarly-0.20.0\dist\extension\index.node.js:3:10348)
	at wo.fireClose (c:\Users\spind\.vscode\extensions\znck.grammarly-0.20.0\dist\extension\index.node.js:3:12622)
	at ChildProcess.<anonymous> (c:\Users\spind\.vscode\extensions\znck.grammarly-0.20.0\dist\extension\index.node.js:30:11847)
	at ChildProcess.emit (node:events:525:35)
	at maybeClose (node:internal/child_process:1091:16)
	at ChildProcess._handle.onexit (node:internal/child_process:302:5)
```

**Comment by gijswijs** [comment]
I suffer from the same issue.

**Comment by doormalena** [comment]
I am having the same issue. Could it be related to HTTPS protocol configuration?

**Comment by jsulopzs** [comment]
I am having the same issue.

```bash
Ready!
Initialized!
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#W0sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#W1sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#W2sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#W3sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#W4sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#W5sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#W6sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X10sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X11sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X12sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X13sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X14sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X15sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X16sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X20sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X21sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X22sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X23sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X24sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X25sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X26sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X30sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X31sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X32sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X33sZmlsZQ%3D%3D
open vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#X34sZmlsZQ%3D%3D
create text checking session for "vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#W0sZmlsZQ%3D%3D" with {
  "documentDialect": "auto-text",
  "documentDomain": "general",
  "suggestions": {
    "ConjunctionAtStartOfSentence": false,
    "Fluency": true,
    "InformalPronounsAcademic": false,
    "MissingSpaces": true,
    "NounStrings": true,
    "NumbersBeginningSentences": true,
    "NumbersZeroThroughTen": true,
    "OxfordComma": false,
    "PassiveVoice": false,
    "PersonFirstLanguage": true,
    "PossiblyBiasedLanguageAgeRelated": true,
    "PossiblyBiasedLanguageDisabilityRelated": true,
    "PossiblyBiasedLanguageFamilyRelated": true,
    "PossiblyBiasedLanguageGenderRelated": true,
    "PossiblyBiasedLanguageHumanRights": true,
    "PossiblyBiasedLanguageHumanRightsRelated": true,
    "PossiblyBiasedLanguageLgbtqiaRelated": true,
    "PossiblyBiasedLanguageRaceEthnicityRelated": true,
    "PossiblyPoliticallyIncorrectLanguage": true,
    "PrepositionAtTheEndOfSentence": false,
    "PunctuationWithQuotation": true,
    "ReadabilityFillerwords": true,
    "ReadabilityTransforms": true,
    "SentenceVariety": true,
    "SpacesSurroundingSlash": true,
    "SplitInfinitive": true,
    "StylisticFragments": false,
    "UnnecessaryEllipses": false,
    "Variety": true,
    "Vocabulary": true
  }
} 
ready vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#W0sZmlsZQ%3D%3D
connecting vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#W0sZmlsZQ%3D%3D
create text checking session for "vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#W1sZmlsZQ%3D%3D" with {
  "documentDialect": "auto-text",
  "documentDomain": "general",
  "suggestions": {
    "ConjunctionAtStartOfSentence": false,
    "Fluency": true,
    "InformalPronounsAcademic": false,
    "MissingSpaces": true,
    "NounStrings": true,
    "NumbersBeginningSentences": true,
    "NumbersZeroThroughTen": true,
    "OxfordComma": false,
    "PassiveVoice": false,
    "PersonFirstLanguage": true,
    "PossiblyBiasedLanguageAgeRelated": true,
    "PossiblyBiasedLanguageDisabilityRelated": true,
    "PossiblyBiasedLanguageFamilyRelated": true,
    "PossiblyBiasedLanguageGenderRelated": true,
    "PossiblyBiasedLanguageHumanRights": true,
    "PossiblyBiasedLanguageHumanRightsRelated": true,
    "PossiblyBiasedLanguageLgbtqiaRelated": true,
    "PossiblyBiasedLanguageRaceEthnicityRelated": true,
    "PossiblyPoliticallyIncorrectLanguage": true,
    "PrepositionAtTheEndOfSentence": false,
    "PunctuationWithQuotation": true,
    "ReadabilityFillerwords": true,
    "ReadabilityTransforms": true,
    "SentenceVariety": true,
    "SpacesSurroundingSlash": true,
    "SplitInfinitive": true,
    "StylisticFragments": false,
    "UnnecessaryEllipses": false,
    "Variety": true,
    "Vocabulary": true
  }
} 
ready vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#W1sZmlsZQ%3D%3D
connecting vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#W1sZmlsZQ%3D%3D
create text checking session for "vscode-notebook-cell:/Users/sota/Dropbox/02%20Company/1%20Projects/datons.academy/Products/datons.stories/01_Challenges%20%26%20Posts/03_Topics/Energy/Spanish%20Energy%20Market/01_Spanish%20Market%20Overview/01_Capacity/en/article/script.ipynb#W2sZmlsZQ%3D%3D" with {
  "documentDialect": "auto-text",
  "documentDomain": "general",
  "suggestions": {
    "ConjunctionAtStartOfSentence": false,
    "Fluency": true,
    "InformalPronounsAcademic": false,
    "MissingSpaces": true,
    "NounStrings": true,
    "NumbersBeginningSentences": true,
    "NumbersZeroThroughTen": true,
    "OxfordComma": false,
    "PassiveVoice": false,
    "PersonFirstLanguage": true,
    "PossiblyBiasedLanguageAgeRelated": true,
    "PossiblyBiasedLanguageDisabilityRelated": true,
    "PossiblyBiasedLanguageFamilyRelated": true,
    "PossiblyBiasedLanguageGenderRelated": true,
    "PossiblyBiasedLanguageHumanRights": true,
    "PossiblyBiasedLanguageHumanRightsRelated": true,
    "PossiblyBiasedLanguageLgbtqiaRelated": true,
    "PossiblyBiasedLanguageRaceEthnicityRelated": true,
    "PossiblyPoliticallyIncorrectLanguage": true,
    "PrepositionAtTheEndOfSentence": false,
    "PunctuationWithQuotation": true,
    "ReadabilityFillerwords": true,
    "ReadabilityTransforms": true,
    "SentenceVariety": true,
    "SpacesSurroundingSlash": true,
    "SplitInfinitive": true,
    "StylisticFragments": false,
    "UnnecessaryEllipses": false,
    "Variety": true,
    "Vocabulary": true
  }
} 
node:internal/deps/undici/undici:11413
    Error.captureStackTrace(err, this);
          ^

TypeError: Failed to parse URL from /Users/sota/.vscode/extensions/znck.grammarly-0.22.1/dist/server/tree-sitter.wasm
    at Object.fetch (node:internal/deps/undici/undici:11413:11) {
  [cause]: TypeError [ERR_INVALID_URL]: Invalid URL
      at new NodeError (node:internal/errors:399:5)
      at URL.onParseError (node:internal/url:566:9)
      at new URL (node:internal/url:646:5)
      at new Request (node:internal/deps/undici/undici:7002:25)
      at fetch2 (node:internal/deps/undici/undici:10554:25)
      at Object.fetch (node:internal/deps/undici/undici:11411:18)
      at fetch (node:internal/process/pre_execution:248:25)
      at /Users/sota/.vscode/extensions/znck.grammarly-0.22.1/dist/server/index.node.js:34:77333
      at /Users/sota/.vscode/extensions/znck.grammarly-0.22.1/dist/server/index.node.js:34:77553
      at new Promise (<anonymous>) {
    input: '/Users/sota/.vscode/extensions/znck.grammarly-0.22.1/dist/server/tree-sitter.wasm',
    code: 'ERR_INVALID_URL'
  }
}

Node.js v18.15.0
[Error - 14:44:55] Connection to server got closed. Server will not be restarted.
[Error - 14:44:55] Request textDocument/codeAction failed.
Error: Connection got disposed.
    at Object.dispose (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-jsonrpc@6.0.0/node_modules/vscode-jsonrpc/lib/common/connection.js:1044:27)
    at Object.dispose (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-languageclient@7.0.0/node_modules/vscode-languageclient/lib/common/client.js:73:35)
    at os.handleConnectionClosed (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-languageclient@7.0.0/node_modules/vscode-languageclient/lib/common/client.js:2424:42)
    at os.handleConnectionClosed (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-languageclient@7.0.0/node_modules/vscode-languageclient/lib/node/main.js:155:15)
    at r (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-languageclient@7.0.0/node_modules/vscode-languageclient/lib/common/client.js:2411:18)
    at Zs.invoke (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-jsonrpc@6.0.0/node_modules/vscode-jsonrpc/lib/common/events.js:55:39)
    at Kr.fire (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-jsonrpc@6.0.0/node_modules/vscode-jsonrpc/lib/common/events.js:117:36)
    at Rt (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-jsonrpc@6.0.0/node_modules/vscode-jsonrpc/lib/common/connection.js:256:26)
    at Zs.invoke (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-jsonrpc@6.0.0/node_modules/vscode-jsonrpc/lib/common/events.js:55:39)
    at Kr.fire (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-jsonrpc@6.0.0/node_modules/vscode-jsonrpc/lib/common/events.js:117:36)
    at wo.fireClose (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-jsonrpc@6.0.0/node_modules/vscode-jsonrpc/lib/common/messageReader.js:40:27)
    at ChildProcess.<anonymous> (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-jsonrpc@6.0.0/node_modules/vscode-jsonrpc/lib/node/main.js:33:45)
    at ChildProcess.emit (node:events:525:35)
    at maybeClose (node:internal/child_process:1091:16)
    at Socket.<anonymous> (node:internal/child_process:449:11)
    at Socket.emit (node:events:513:28)
    at Pipe.<anonymous> (node:net:322:12)
[Error - 14:44:55] Request textDocument/codeAction failed.
Error: Connection got disposed.
    at Object.dispose (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-jsonrpc@6.0.0/node_modules/vscode-jsonrpc/lib/common/connection.js:1044:27)
    at Object.dispose (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-languageclient@7.0.0/node_modules/vscode-languageclient/lib/common/client.js:73:35)
    at os.handleConnectionClosed (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-languageclient@7.0.0/node_modules/vscode-languageclient/lib/common/client.js:2424:42)
    at os.handleConnectionClosed (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-languageclient@7.0.0/node_modules/vscode-languageclient/lib/node/main.js:155:15)
    at r (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-languageclient@7.0.0/node_modules/vscode-languageclient/lib/common/client.js:2411:18)
    at Zs.invoke (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-jsonrpc@6.0.0/node_modules/vscode-jsonrpc/lib/common/events.js:55:39)
    at Kr.fire (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-jsonrpc@6.0.0/node_modules/vscode-jsonrpc/lib/common/events.js:117:36)
    at Rt (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-jsonrpc@6.0.0/node_modules/vscode-jsonrpc/lib/common/connection.js:256:26)
    at Zs.invoke (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-jsonrpc@6.0.0/node_modules/vscode-jsonrpc/lib/common/events.js:55:39)
    at Kr.fire (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-jsonrpc@6.0.0/node_modules/vscode-jsonrpc/lib/common/events.js:117:36)
    at wo.fireClose (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-jsonrpc@6.0.0/node_modules/vscode-jsonrpc/lib/common/messageReader.js:40:27)
    at ChildProcess.<anonymous> (/Users/sota/.vscode/extensions/node_modules/.pnpm/vscode-jsonrpc@6.0.0/node_modules/vscode-jsonrpc/lib/node/main.js:33:45)
    at ChildProcess.emit (node:events:525:35)
    at maybeClose (node:internal/child_process:1091:16)
    at Socket.<anonymous> (node:internal/child_process:449:11)
    at Socket.emit (node:events:513:28)
    at Pipe.<anonymous> (node:net:322:12)
```

**Comment by doormalena** [comment]
I can confirm it relates to the Node.Js version packaged within VS Code. See also https://code.visualstudio.com/updates/v1_82#_electron-25-update

Rolling back to a version older than v1.82.0 works (e.g. v1.81.1).

**Comment by miohtama** [comment]
For the reference you can download Visual Studio Code v1.81.0 here 

https://code.visualstudio.com/updates/v1_81

**Comment by KedneckInc** [comment]
Confirm the rollback to VSC v1.81.1 works.

**Comment by AndrewFarley** [comment]
Please fix.  :(  Rolled back VS code for now, sadly.

**Comment by wagslane** [comment]
I had to roll back as well, would be awesome if this worked with the lastest version....

**Comment by KedneckInc** [comment]
Unfortunately, I don't think it will ever be fixed.  Grammarly is intent on removing the API that this project depends on.  I have argued against it as a paying customer who depends on this API, but they don't seem interested.  On the flip side, they do have a Windows client that works through the accessibility interfaces.  Regrettably, VSCode does not appear to use a variant of the accessibility interfaces that Grammarly can work with.  Works fine with things like Notepad, but I'm not going to write my stories in that archaic tool.

**Comment by Samreay** [comment]
@KedneckInc if you find other tools that make VSCode a nicer place to write, please let me know.

**Comment by KedneckInc** [comment]
@Samreay Since I write in Markdown, I've found "Markdown All in One" quite useful.  I've tried tools like Wavemaker, but I don't like the constraints of the interface.  If Grammarly stops working altogether in VSC, I may not have a choice.

I'm an old emacs hand, and not having my emacs style editing keymap is gonna put a crimp in my writing.  Not to mention most such tools don't support markdown natively.

**Comment by AndrewFarley** [comment]
If Grammarly is (idiotically) removing their API for this, does someone know a good alternative similar service?

**Comment by yogidevendra** [comment]
Is it sustainable to consider other options such as https://github.com/languagetool-org/languagetool for integration instead of relying on Grammerly developer API which is scheduled for deprecation.

**Comment by ahayman** [comment]
With Grammarly deprecating their public API, this issue is a good excuse to move away from Grammarly.  As @yogidevendra mentioned, LanguageTool is probably the best free option.  However, I found a few defunct LanguageTool VSCode extensions.  This one in particular doesn't work (for me at least) but shows up near the top of the list: https://marketplace.visualstudio.com/items?itemName=adamvoss.vscode-languagetool.  It has not been updated since 2018.  I do *not* recommend it.

I found this one, which does seem to work: https://marketplace.visualstudio.com/items?itemName=valentjn.vscode-ltex.  You can find it by searching for LTEX.  

Advantages:
 - It does a good job of spelling errors.
 - It's decent at grammar, but not nearly as good as Grammarly (or other paid tools).  However, it catches most of the common grammar mistakes.
 - You can add words to the spelling dictionary.  This is something I could never get working in Grammarly, even if I put the word in through their website and ensured I had signed in through the extension.
 - It does everything locally (no API calls), which means it's quicker and will work offline.

Disadvantages:
 - It's just not as comprehensive as Grammarly. While it supports basic grammar errors, it doesn't even approach what the free tier of Grammarly can do, much less their paid tiers. 

Switching away from Grammarly isn't a huge loss for me personally.  Grammarly always tended toward "essay-ish" styles of writing, whereas I use VSCode to write fiction. I found myself ignoring a lot of Grammarly's suggestions for this reason. At least with LanguageTool, I almost always want to fix the errors it surfaces.  Even if it doesn't catch as many errors, I end up with a more pleasant writing experience—no more ignoring a bunch of grammar errors.

**Comment by dylan-k** [comment]
[Vale ](https://vale.sh/)is another potential replacement, although some setup is required. I use it along with the [VScode extension for vale](https://marketplace.visualstudio.com/items?itemName=ChrisChinchilla.vale-vscode). It's free, open source, regularly updated and powerful. What I like about it is that I can configure custom style guides for the way I write, with variations for writing in different contexts or styles. Over time, using vale, i've got my own personalize in-house styleguide right inside of VScode.

**Comment by znck** [comment]
Fixed today in v0.24.0
